diff --git a/src/flash/nor/kinetis.c b/src/flash/nor/kinetis.c
index 48a5de4..c022177 100644
--- a/src/flash/nor/kinetis.c
+++ b/src/flash/nor/kinetis.c
@@ -234,6 +234,7 @@
 #define KINETIS_SDID_PROJECTID_MASK  KINETIS_SDID_DIEID_MASK
 #define KINETIS_SDID_PROJECTID_KE1xF 0x00000080
 #define KINETIS_SDID_PROJECTID_KE1xZ 0x00000100
+#define KINETIS_SDID_PROJECTID_S32K  0x00000000

 struct kinetis_flash_bank {
 	struct kinetis_chip * k_chip;
@@ -2398,6 +2399,18 @@ static int kinetis_probe_chip(struct kinetis_chip * k_chip)
 				snprintf(name, sizeof(name), "MKE%u%uF%%s%u",
 					 familyid, subfamid, cpu_mhz / 10);
 				break;
+                        case KINETIS_SDID_FAMILYID_K1X | KINETIS_SDID_SUBFAMID_KX8 | KINETIS_SDID_PROJECTID_S32K:
+				/* KE1xF: FTFE, 4kB sectors */
+				k_chip->pflash_sector_size = 4<<10;
+				k_chip->nvm_sector_size = 2<<10;
+				k_chip->max_flash_prog_size = 1<<10;
+				num_blocks = 2;
+				k_chip->flash_support = FS_PROGRAM_PHRASE | FS_PROGRAM_SECTOR;
+				k_chip->cache_type = KINETIS_CACHE_MSCM;
+
+				cpu_mhz = 80;
+				snprintf(name, sizeof(name), "S32K144HFVLL");
+				break;

 			default:
 				LOG_ERROR("Unsupported KE FAMILYID SUBFAMID");
